Create log configuration functions

CaioCordeiro opened the pull request:
Changes:

- Create src/log.c and src/log.h and Add them to src/Makefile.am;
- Create ell lib configuration functions;
- Include log.h in knot_cloud.c;
- Setup log configurations inside knot_cloud_start function;
- Create knot_cloud_set_log_priority, this function lets the user
choose the SDK log priority level;
- Add function to knot_cloud.h

luigiluz commented on src/log.c:
```diff
@@ -0,0 +1,68 @@
+/*
+ * This file is part of the KNOT Project
+ *
+ * Copyright (c) 2019, CESAR. All rights reserved.
```
```suggestion
 * Copyright (c) 2020, CESAR. All rights reserved.
```

luigiluz commented on src/log.c:
```diff
@@ -0,0 +1,68 @@
+    log_priority = priority;    
+    if (log_priority == L_LOG_DEBUG)
+		l_debug_enable("*");
+}
```
Please add a blank line at the end of the file

luigiluz commented on src/log.c:
```diff
@@ -0,0 +1,68 @@
+ *  Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ */
+
```
```suggestion

/**
 *  Log source file
 */
 
```

luigiluz commented on src/log.h:
```diff
@@ -0,0 +1,23 @@
+/*
+ * This file is part of the KNOT Project
+ *
+ * Copyright (c) 2019, CESAR. All rights reserved.
```
```suggestion
 * Copyright (c) 2020, CESAR. All rights reserved.
```

luigiluz commented on src/log.h:
```diff
@@ -0,0 +1,23 @@
+ *  Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ */
+
```
```suggestion

/**
 *  Log header file
 */
 
```

luigiluz commented on src/log.h:
```diff
@@ -0,0 +1,23 @@
+ */
+
+void log_set_priority(int priority);
+void log_ell_enable();
```
Please add a blank line at the end of the file

alla3 commented on src/knot_cloud.c:
```diff
@@ -723,6 +723,18 @@ int knot_cloud_start(char *url, char *user_token,
 
+void knot_cloud_set_log_priority(char * priority)
+{
+    if (!strcmp(priority,"error"))
```
This indentation is using spaces instead of tab. Please, change it.

alla3 commented on src/knot_cloud.c:
```diff
@@ -723,6 +723,18 @@ int knot_cloud_start(char *url, char *user_token,
+		log_set_priority(L_LOG_INFO);
+	else if (!strcmp(priority,"debug"))
+		log_set_priority(L_LOG_DEBUG);
+}
```
This function is not following the same order of the header.

alla3 commented on src/log.c:
```diff
@@ -0,0 +1,72 @@
+	l_log_set_handler(log_stderr_handler);
+}
+
+void log_set_priority(int priority)
+{
+    log_priority = priority;    
+    if (log_priority == L_LOG_DEBUG)
+		l_debug_enable("*");
+}
```
I would add this function on commit 9944be4700a2e2088b1afc89e85da0204767ffd1, since it is not been used on this commit.

CaioCordeiro committed:
Create log configurarion functions

Changes:

- Create src/log.c and src/log.h;
- Add them to src/Makefile.am;
- Create ell lib configuration functions;
- Include log.h in knot_cloud.c;
- Setup log configurations inside knot_cloud_start function;

alla3 commented on src/knot_cloud.c:
```diff
@@ -723,6 +723,18 @@ int knot_cloud_start(char *url, char *user_token,
 	return mq_start(url, connected_cb, disconnected_cb, user_data);
 }
 
+void knot_cloud_set_log_priority(char * priority)
```
```suggestion
void knot_cloud_set_log_priority(char *priority)
```

alla3 commented on src/knot_cloud.h:
```diff
@@ -52,12 +52,14 @@ typedef bool (*knot_cloud_cb_t) (const struct knot_cloud_msg *msg,
 		     knot_cloud_connected_cb_t connected_cb,
 		     knot_cloud_disconnected_cb_t disconnected_cb,
 		     void *user_data);
+void knot_cloud_set_log_priority(char * priority);
```
```suggestion
void knot_cloud_set_log_priority(char *priority);
```

alla3 commented on src/log.c:
```diff
@@ -63,3 +63,9 @@ void log_ell_enable()
 {
 	l_log_set_handler(log_stderr_handler);
 }
+void log_set_priority(int priority)
```
Please, add a blank line between functions.

vitbaq replied:
Put a space before this line

vitbaq commented on src/log.c:
```diff
@@ -63,3 +63,10 @@ void log_ell_enable()
 	l_log_set_handler(log_stderr_handler);
 }
+
+void log_set_priority(int priority)
```
Change the order of the functions. Put this function after the log_ell_enable

vitbaq commented on src/log.c:
```diff
@@ -0,0 +1,65 @@
+	vfprintf(stderr, format, ap);
+}
+
+void log_ell_enable()
```
Change to `void log_ell_enable(void)`

vitbaq replied:
Reason for this: [link](https://softwareengineering.stackexchange.com/questions/286490/what-is-the-difference-between-function-and-functionvoid/287002)

vitbaq commented on src/log.h:
```diff
@@ -0,0 +1,26 @@
+ * Log header file
+ */
+
+void log_ell_enable();
```
Change to `void log_ell_enable(void)`

vitbaq reviewed (changes requested):
Almost there, the last things to change.

CaioCordeiro committed:
Create set priority function

Changes:

- Create set_log_priority inside log.c;
- Create knot_cloud_set_log_priority, this function lets the user
choose the SDK log priority level;
- Add function to knot_cloud.h;
